packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -65,6 +65,7 @@ import {
 import getComponentName from 'shared/getComponentName';
 import invariant from 'shared/invariant';
 import warningWithoutStack from 'shared/warningWithoutStack';
+import {canUseDOM} from 'shared/ExecutionEnvironment';
 
```

Comment:
It feels wrong to have any logic concerning the DOM (even checks) in the reconciler. It’s also why the warning false positives for the test renderer.

We should somehow make it specific to ReactDOM I think.

Reply:
I’d originally put the dom check inside ReactFiberHooks, I’ll move it back there. I’m thinking I’ll test the fiber’s internal instance and trst if it’s a dom element.

Reply:
ReactFiberHooks isn’t any better. It’s still in the reconciler. 

I meant to add a host config method that renderers can implement. Look at how server hydration warnings are done.